app/scripts/modules/netflix/fastProperties/scope/fastPropertyScopeSearchCategory.service.ts

```diff
@@ -0,0 +1,258 @@
+        serverGroupList.forEach((serverGroup) => {
+          if (serverGroup.region === scope.region) {
+            acc.push(serverGroup.instanceCounts);
+          };
+        });
+        return acc;
+      }, [])
+      .reduce((acc: any, counts: any) => {
+        return _.mergeWith(acc, counts, (a: number, b: number) =>  a + b );
+      }, scope.instanceCounts);
+
+    return serverGroupInstanceCounts;
+  }
+
+
+  /**
+   * Builds Cluster level fast property scope
+   */
+  private buildScopeForCluster(selected: any, applicationDictionary: any): Scope[] {
+    let application = applicationDictionary[selected.application];
+    let scope = new Scope();
+    scope.appId = selected.application;
+    scope.cluster = selected.cluster;
+    scope.stack = selected.stack;
+    scope.instanceCounts = this.impactCountForCluster(scope, application);
+    return [scope];
+  }
+
+  public impactCountForCluster(scope: Scope, application: any): any {
+    let foundCluster: any = application.clusters
+      .filter((cluster: any) => cluster.name === scope.cluster)
+      .shift();
+
+    return foundCluster ? foundCluster.instanceCounts : scope.instanceCounts;
+  }
+
+
+  /**
+   * Builds Server Group level fast property scope
+   */
+  private buildScopeForServerGroup(selected: any, applicationDictionary: any): Scope[] {
+    let application = applicationDictionary[selected.application];
+    let scope = new Scope();
+    scope.appId = selected.application;
+    scope.cluster = selected.cluster;
+    scope.stack = selected.stack;
+    scope.region = selected.region;
+    scope.asg = selected.serverGroup;
+    scope.instanceCounts = this.impactCountForServerGroup(scope, application);
+    return [scope];
+  }
+
+  public impactCountForServerGroup(scope: Scope, application: Application): any {
+    let foundCluster = application.clusters
+      .filter((cluster: any) => cluster.name === scope.cluster)
+      .shift();
+
+    if (foundCluster && foundCluster.serverGroups) {
+      let serverGroup = foundCluster.serverGroups
+        .filter((serverGroup: any) => serverGroup.name === scope.asg)
+        .shift();
+
+      return serverGroup ? serverGroup.instanceCounts : scope.instanceCounts;
+    };
+
+    return scope.instanceCounts;
+  }
+
+
+  /**
+   * Builds Instance level fast property scope
+   */
+  private buildScopeForInstance(selected: any): Scope[] {
+    let scope = new Scope();
+    scope.appId = selected.application;
+    scope.cluster = selected.cluster;
+    scope.region = selected.region;
+    scope.asg = selected.serverGroup;
+    scope.serverId = selected.instanceId;
+    scope.instanceCounts = this.impactCountForInstnace();
+    return [scope];
+  }
+
+  private impactCountForInstnace() {
+    return {up: 1}; // hard code it assume the instance is up if not ¯\_(ツ)_/¯
+  }
+
+  public excludeUnneededCategories(results: any[]): any[] {
+    const notNeeded = ['Projects', 'Security Groups', 'Load Balancers'];
```

Comment:
Since the list of search items is likely to grow in the future (libraries, repositories, etc), it might make more sense to whitelist the results we care about, rather than blacklist the ones we don't.